src/ios/GeolocationPlugin.m

```diff
@@ -0,0 +1,314 @@
+    
+    //[self MakeJSON_Make];
+    
+    NSLog(@"Now monitoring for %@", region.identifier);
+    NSLog(@"Welcome to %f", region.radius);
+    NSString *message = @"Show Message...\n didStartMonitoringForRegion";
+    UIAlertView *toast = [[UIAlertView alloc]initWithTitle:nil message:message delegate:nil cancelButtonTitle:nil otherButtonTitles:nil, nil];
+    
+    [toast show];
+    int duration = 1;
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, duration *NSEC_PER_SEC), dispatch_get_main_queue(), ^{[toast dismissWithClickedButtonIndex:0 animated:YES];
+    });
+}
+-(void)locationManager:(CLLocationManager *)manager didEnterRegion:(CLRegion *)region{
+    EnterRegion = @"YES";
+    
+    NSString *message = @"Show Message...\n Called didEnterRegion function";
+    UIAlertView *toast = [[UIAlertView alloc]initWithTitle:nil message:message delegate:nil cancelButtonTitle:nil otherButtonTitles:nil, nil];
+    
+    [toast show];
+    int duration = 1;
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, duration *NSEC_PER_SEC), dispatch_get_main_queue(), ^{[toast dismissWithClickedButtonIndex:0 animated:YES];
+    });
+    
+}
+-(void)locationManager:(CLLocationManager *)manager didExitRegion:(CLRegion *)region{
+    EnterRegion = @"NO";
+    
+    NSString *message = @"Show Message...\n Called didExitRegion function";
+    UIAlertView *toast = [[UIAlertView alloc]initWithTitle:nil message:message delegate:nil cancelButtonTitle:nil otherButtonTitles:nil, nil];
+    
+    [toast show];
+    int duration = 1;
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, duration *NSEC_PER_SEC), dispatch_get_main_queue(), ^{[toast dismissWithClickedButtonIndex:0 animated:YES];
+    });
+}
+
+-(void)CreateDataBase{
+    
+    //Get the documents directory
+    NSArray *dirPaths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
+    NSString *docsDir = [dirPaths objectAtIndex:0];
+    
+    //Build the path to the database file
+    dataBasePath = [[NSString alloc]initWithString:[docsDir stringByAppendingPathComponent:@"GPS_Track.sqlite"]];
+    
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+    
+    
+    
+    if ([fileManager fileExistsAtPath:dataBasePath] == NO) {
+        const char *dbpath = [dataBasePath UTF8String];
+        
+        if (sqlite3_open(dbpath, &contactDB) == SQLITE_OK) {
+            char *errMsg;
+            
+            const char *sql_table = "CREATE TABLE GPSTable(_id INTEGER PRIMARY KEY NOT NULL, latitude TEXT, longitude TEXT, altitude TEXT, timerInterval TEXT, region TEXT);";
+            
+            if (sqlite3_exec(contactDB, sql_table, NULL, NULL, &errMsg) != SQLITE_OK) {
+                NSLog(@"Failed to create table");
+            }
+            sqlite3_close(contactDB);
+        }
+        else{
+            NSLog(@"Failed to open/create databese");
+        }
+    }
+}
+
+-(void)InsertData{
```

Comment:
this method uses global variables as the source of the data it inserts. using global variables in multi-threaded code is never a good idea as multiple events can be happening at the same time. please for all methods always pass the data via method arguments, e.g. something like `insertData:(SomeDataObject *)data` or `insertData:text, Lat:lat, Lon:lon, Date:date)` ...